### PR TITLE
add GetAppliedIndex rpc

### PR DESCRIPTION
**Pull request by TennyZhuang**

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -562,3 +562,11 @@ message ReadIndexResponse{
     uint64 read_index = 2;
 }
 
+message GetAppliedIndexRequest {
+    uint64 region_id = 1;
```
Comment (hicqu) [inline proto/kvrpcpb.proto]: How about add a new service `MultiDCProxy`, and put a new RPC `GetCommittedIndexAndTs` in it? BTW we need to contains a `Context` in the request.

File: proto/tikvpb.proto [hunk 2]
```diff
@@ -61,6 +61,7 @@ service Tikv {
     // Region commands.
     rpc SplitRegion (kvrpcpb.SplitRegionRequest) returns (kvrpcpb.SplitRegionResponse) {}
     rpc ReadIndex(kvrpcpb.ReadIndexRequest) returns (kvrpcpb.ReadIndexResponse) {}
+    rpc GetCommittedIndexAndTs(kvrpcpb.GetCommittedIndexAndTsRequest) returns (kvrpcpb.GetCommittedIndexAndTsResponse) {}
```
Comment (hicqu) [inline proto/tikvpb.proto]: Without modify `BatchCommandsRequest` you can't use it `BatchCommands`. So I prefer to put it in an new service to make development faster.

File: proto/kvrpcpb.proto [hunk 3]
```diff
@@ -562,3 +563,12 @@ message ReadIndexResponse{
     uint64 read_index = 2;
 }
 
+message GetCommittedIndexAndTsRequest {
+    Context context = 1;
```
Comment (hicqu) [inline proto/kvrpcpb.proto]: Need to be `repeated Context contexts = 1` because one SQL can access many regions.

Comment (hicqu) [reply proto/kvrpcpb.proto]: Maybe we also need a new field `ts_required`. TiKV can leave `ts` as 0 in response if `ts_required` is false.

Comment (TennyZhuang) [reply proto/kvrpcpb.proto]: What is the difference between `ts_required == false` and call ReadIndex?

**Comment by TennyZhuang** [comment]
@hicqu PTAL

**Pull request merged**